felix/fv/policysync_test.go

```diff
@@ -511,18 +515,18 @@ var _ = Context("_POL-SYNC_ _BPF-SAFE_ policy sync API tests", func() {
 
 						It("should sync service account to each workload", func() {
 							for _, c := range mockWlClient {
-								Eventually(c.ServiceAccounts).Should(Equal(map[proto.ServiceAccountID]*proto.ServiceAccountUpdate{
-									saID: {
-										Id:     &saID,
-										Labels: map[string]string{"key.1": "value.1", "key_2": "value-2"},
-									},
-								}))
+								v := c.ServiceAccounts()
+								equal := googleproto.Equal(v[saID], &proto.ServiceAccountUpdate{
+									Id:     types.ServiceAccountIDToProto(saID),
+									Labels: map[string]string{"key.1": "value.1", "key_2": "value-2"},
+								})
+								Expect(equal).To(BeTrue())
```

Comment:
This was an `Eventually` check but now changed to `Expect`. This test may need to check periodically and wait for it to pass (same below).